velox/dwio/dwrf/test/E2EFilterTestBase.cpp

```diff
@@ -411,4 +394,33 @@ void E2EFilterTestBase::testWithTypes(
   }
 }
 
+void E2EFilterTestBase::checkOwnership(VectorPtr batch) {
+  // Check the 6 first pairs of previous, next batch to see that
+  // fetching the next does not overwrite parts reachable from a
+  // retained reference to the previous one.
+  if (batchCounter_ > 11) {
+    return;
+  }
+  if (batchCounter_ % 2 == 0) {
+    previousBatch_ = std::make_shared<RowVector>(
+        batch->pool(),
+        batch->type(),
+        BufferPtr(nullptr),
+        batch->size(),
+        batch->as<RowVector>()->children());
+    previousBatchCopy_ =
+        BaseVector::create(batch->type(), batch->size(), batch->pool());
```

Comment:
It would be nice to add a helper static method BaseVector::copy(const VectorPtr& vector)